File: 169-majority-element/majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        candidate = 0
        vote = 0
        for num in nums:
            if vote == 0:
                candidate = num
                vote += 1
            else:
                if(num == candidate):
                    vote += 1
                else:
                    vote -= 1
        count = 0
        for num in nums:
            if num == candidate:
                count += 1
            if count > len(nums) / 2:
                return candidate
        return -1       

File: 169-majority-element/majority-element.c
int majorityElement(int* nums, int numsSize) 
{
    int candidate = 0;
    int vote = 0;
    for(int i = 0; i < numsSize; i++)
    {
        if(vote == 0)
        {
            candidate = nums[i];
            vote++;
        }
        else
        {
            if(nums[i] == candidate)
                vote++;
            else 
                vote--;
        }
    }
        int count = 0;
        for(int i = 0; i < numsSize; i++)
        {
            if(nums[i] == candidate)
            count++;
        }
        if(count > (numsSize / 2))
        return candidate;
        return -1;
}